qiskit_aqua/translators/ising/exactcover.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 IBM.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+
+
+
+import logging
+from collections import OrderedDict
+
+import numpy as np
+import numpy.random as rand
+from qiskit.quantum_info import Pauli
+
+from qiskit_aqua import Operator
+
+logger = logging.getLogger(__name__)
+
+def random_number_list(n, weight_range=100, savefile=None):
+    """Generate a set of positive integers within the given range.
+
+    Args:
+        n (int): size of the set of numbers.
+        weight_range (int): maximum absolute value of the numbers.
+        savefile (str or None): write numbers to this file.
+
+    Returns:
+        numpy.ndarray: the list of integer numbers.
+    """
+    number_list = np.random.randint(low=1, high=(weight_range+1), size=n)
+    if savefile:
+        with open(savefile, 'w') as outfile:
+            for i in range(n):
+                outfile.write('{}\n'.format(number_list[i]))
+    return number_list
+
+def get_exactcover_qubitops(list_of_subsets):
+    """Construct the Hamiltonian for the exact solver problem
+
+
+    Args:
+        list_of_subsets: list of lists (i.e., subsets)
+
+    Returns:
+        operator.Operator, float: operator for the Hamiltonian and a
+        constant shift for the obj function.
+
+    Assumption:
+        the union of the subsets contains all the elements to cover
+
+    """
+    n = len(list_of_subsets)
+    # The Hamiltonian is:
```

Comment:
the description here can be a part of docstring of function

Reply:
done